components/sst_importer/src/sst_importer.rs

```diff
@@ -490,21 +662,41 @@ impl SstImporter {
         let start = Instant::now();
         let dst_name = format!("{}_{}", meta.get_name(), meta.get_range_offset());
 
-        let mut lock = self
-            .file_locks
-            .entry(dst_name)
-            .or_insert((CacheKvFile::Mem(Arc::default()), Instant::now()));
-
-        if let CacheKvFile::Mem(buff) = &lock.0 {
-            if !buff.is_empty() {
-                lock.1 = Instant::now();
-                return Ok(lock.0.clone());
+        let promise = {
+            let lock = self.file_locks.entry(dst_name);
+            IMPORTER_APPLY_DURATION
+                .with_label_values(&["download-get-lock"])
+                .observe(start.saturating_elapsed().as_secs_f64());
+
+            match lock {
+                Entry::Occupied(mut ent) => match ent.get_mut() {
+                    (CacheKvFile::Mem(buff), last_used) => {
+                        *last_used = Instant::now();
```

Comment:
BTW what's the usage of `last_used`?

Reply:
We would check it in interval and remove outdated cache according to it.